docs/reference/streamfield_reference.rst

```diff
@@ -49,472 +49,442 @@ All block types accept the following optional keyword arguments:
+           class Meta:
+               icon = 'user'
 
-.. code-block:: python
 
-    ('ingredients_list', blocks.ListBlock(blocks.CharBlock(label="Ingredient")))
+   The ``Meta`` class supports the properties ``default``, ``label``, ``icon`` and ``template``, which have the same meanings as when they are passed to the block's constructor.
 
+   This defines ``PersonBlock()`` as a block type for use in StreamField definitions:
 
-Any block type is valid as the sub-block type, including structural types:
+   .. code-block:: python
 
-.. code-block:: python
+       body = StreamField([
+           ('heading', blocks.CharBlock(form_classname="full title")),
+           ('paragraph', blocks.RichTextBlock()),
+           ('image', ImageChooserBlock()),
+           ('person', PersonBlock()),
+       ])
 
-    ('ingredients_list', blocks.ListBlock(blocks.StructBlock([
-        ('ingredient', blocks.CharBlock()),
-        ('amount', blocks.CharBlock(required=False)),
-    ])))
+   The following additional options are available as either keyword arguments or Meta class attributes:
 
-To customise the class name of a ``ListBlock`` as it appears in the page editor, you can specify a ``form_classname`` attribute as a keyword argument to the ``ListBlock`` constructor:
+   :param form_classname: An HTML ``class`` attribute to set on the root element of this block as displayed in the editing interface. Defaults to ``struct-block``; note that the admin interface has CSS styles defined on this class, so it is advised to include ``struct-block`` in this value when overriding. See :ref:`custom_editing_interfaces_for_structblock`.
+   :param form_template: Path to a Django template to use to render this block's form. See :ref:`custom_editing_interfaces_for_structblock`.
+   :param value_class: A subclass of ``wagtail.core.blocks.StructValue`` to use as the type of returned values for this block. See :ref:`custom_value_class_for_structblock`.
 
-.. code-block:: python
-    :emphasize-lines: 4
 
-    ('ingredients_list', blocks.ListBlock(blocks.StructBlock([
-        ('ingredient', blocks.CharBlock()),
-        ('amount', blocks.CharBlock(required=False)),
-    ]), form_classname='ingredients-list'))
+.. class:: wagtail.core.blocks.ListBlock
 
-Alternatively, you can add ``form_classname`` in a subclass's ``Meta``:
+   A block consisting of many sub-blocks, all of the same type. The editor can add an unlimited number of sub-blocks, and re-order and delete them. Takes the definition of the sub-block as its first argument:
 
-.. code-block:: python
-    :emphasize-lines: 6
+   .. code-block:: python
 
-    class IngredientsListBlock(blocks.ListBlock):
-        ingredient = blocks.CharBlock()
-        amount = blocks.CharBlock(required=False)
+       ('ingredients_list', blocks.ListBlock(blocks.CharBlock(label="Ingredient")))
```

Comment:
If we're going to use the tuple format as an example, could we keep the ``body = StreamField([ ... ])`` around it?